### PR TITLE
Release/2022 02 08 silver fox

### LINKED ISSUE #2235: Support loading state for asset picker in send modal
There seems to be an asset drop down issue on the fresh install. It works but not right after we install the extension all my automated Stella test are failing because the asset is not available in the drop down after a fresh install and clearing the cache.

This goes away after clicking through the app there seems to be something related to caching. The amount of time it takes to become clickable varies I have seen it take over a minute and I couldn't figure out why the drop down wasn't working in chrome. As you can see in the video in Brave it was a brief amount of time.

https://screencast-o-matic.com/watch/c3n2opVbjvb

### PR DESCRIPTION
**Pull request by beguene**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/1852493771).

Content:

- [feat: add message in activity list](https://github.com/hirosystems/stacks-wallet-web/commit/f67f7ab8aeb3209513e9479405700fe6ca627acd) closes https://github.com/hirosystems/stacks-wallet-web/issues/2187

- [fix: remove suspense, causes scroll bug,](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) [closes](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) https://github.com/hirosystems/stacks-wallet-web/issues/2220

-   Asset dropdown selection issue upon clearing cache https://github.com/hirosystems/stacks-wallet-web/issues/2235

- [fix: add release rules to commit analyzer](https://github.com/hirosystems/stacks-wallet-web/commit/20806261e65dc49db34041a708e786409d520ffb) 
No need to QA this

- [fix: double scrollbar issue firefox,](https://github.com/hirosystems/stacks-wallet-web/pull/2224/commits/75f60a75f60a25605670522fb55c00d536168620) [closes](https://github.com/hirosystems/stacks-wallet-web/pull/2224/commits/75f60a75f60a25605670522fb55c00d536168620) https://github.com/hirosystems/stacks-wallet-web/issues/2228

- [fix: slow fetching of metadata, hit api faster](https://github.com/hirosystems/stacks-wallet-web/pull/2224/commits/0b5fb744853b67c8570e0a3f6f2f7b3cd4f5782f) closes #2235

### CODE REVIEW

**Review by kyranjamie — APPROVED** [review]

**Review by fbwoolf — APPROVED** [review]

**Comment by Eshwari007** [comment]
Tested all PR above once again in Silver fox regression:
[feat: add message in activity list](https://github.com/hirosystems/stacks-wallet-web/commit/f67f7ab8aeb3209513e9479405700fe6ca627acd) closes https://github.com/hirosystems/stacks-wallet-web/issues/2187 (Passed)

[fix: remove suspense, causes scroll bug,](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) [closes](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) https://github.com/hirosystems/stacks-wallet-web/issues/2220 (Passed)

Fix bns lookupProfile
https://github.com/hirosystems/stacks-wallet-web/pull/2089
The fix didnt work for me though @pradel indicated it worked for him.
But test passed for bns lookup only when I tested it against https://github.com/hirosystems/stacks-wallet-web/pull/2172
This was tested on https://sigle-9i6eqj7dj-sigle.vercel.app/ 


cc @kyranjamie @beguene @andresgalante

**Comment by andresgalante** [comment]
How can we test the BNS PR to release with confidence? What are the next steps here?

**Comment by pradel** [comment]
I can confirm that it's not working on my side now, will spend some time today to see what's going on and report my findings here.

**Comment by pradel** [comment]
After some testing, there is something really strange going on in the wallet. I managed to reproduce the issue in a consistent way.

A. User path that is working to login successfully, basically any user installing the extension for the first time.
A.1. Logout from any existing account in the Hiro wallet
A.2 Quit chrome completely - ⚠️ this is really important, you need to make sure the full chrome app is closed and no background process are running.
A.3. Open chrome and login into the Hiro wallet with an account owning a .btc username.
A.4. Go to the Sigle demo app, then login
A.5. ✅  Everything is working you are logged in

B. How to reproduce the issue in a consistent way, with this path you will get the issue @Eshwari007 faced
B.1. Open chrome then login into the Hiro wallet with an account owning a .btc username.
B.2. Logout from the Hiro wallet
B.3. Now login again in the Hiro wallet without closing chrome
B.4. Go to the Sigle demo app, then login
B.4. 🛑  You should get the error @Eshwari007 had before

So basically logging in to a new account is working only if you did not log out before (fresh installation, or restarting chrome after logout). Like this, the issue seems more related to how the wallet handles things internally than in the SDK but might be wrong :).

**Comment by pradel** [comment]
I run the same tests on https://github.com/hirosystems/stacks-wallet-web/pull/2172 and all the tests are passing there. So the bug is really likely in the stacks-wallet-web code.

**Comment by Eshwari007** [comment]
> The fix didnt work for me though @pradel indicated it worked for him. But test passed for bns lookup only when I tested it against #2172 This was tested on https://sigle-9i6eqj7dj-sigle.vercel.app/

Exactly, that was my findings as well as mentioned above
https://github.com/hirosystems/stacks-wallet-web/pull/2224#issuecomment-1033103139
cc @pradel @andresgalante @kyranjamie

**Comment by Eshwari007** [comment]
> After some testing, there is something really strange going on in the wallet. I managed to reproduce the issue in a consistent way.
> 
> A. User path that is working to login successfully, basically any user installing the extension for the first time. A.1. Logout from any existing account in the Hiro wallet A.2 Quit chrome completely - ⚠️ this is really important, you need to make sure the full chrome app is closed and no background process are running. A.3. Open chrome and login into the Hiro wallet with an account owning a .btc username. A.4. Go to the Sigle demo app, then login A.5. ✅ Everything is working you are logged in
> 
> B. How to reproduce the issue in a consistent way, with this path you will get the issue @Eshwari007 faced B.1. Open chrome then login into the Hiro wallet with an account owning a .btc username. B.2. Logout from the Hiro wallet B.3. Now login again in the Hiro wallet without closing chrome B.4. Go to the Sigle demo app, then login B.4. 🛑 You should get the error @Eshwari007 had before
> 
> So basically logging in to a new account is working only if you did not log out before (fresh installation, or restarting chrome after logout). Like this, the issue seems more related to how the wallet handles things internally than in the SDK but might be wrong :).

I'll add this use case to testing in regression as well.
Thank you for the findings @pradel

**Comment by Eshwari007** [comment]
@beguene Seems like the info message background issue resurfaced again
Screen Shot 2022-02-10 at 1 19 01 PM

**Comment by Eshwari007** [comment]
[feat: add message in activity list](https://github.com/hirosystems/stacks-wallet-web/commit/f67f7ab8aeb3209513e9479405700fe6ca627acd) closes https://github.com/hirosystems/stacks-wallet-web/issues/2187.   -**`Background color issue resurfaced ,attached screenshot above`**

[fix: remove suspense, causes scroll bug,](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) [closes](https://github.com/hirosystems/stacks-wallet-web/commit/664a3aeb44993d0d5b10a4323c42371951bf47cb) https://github.com/hirosystems/stacks-wallet-web/issues/2220. -**`Test Passed`**

cc @beguene @timstackblock @andresgalante

**Comment by fbwoolf** [comment]
@beguene I reset the commit to be before the bns commits not realizing you had a fix commit after. So, I would assume the best path forward would be to reset this back to your fix commit and revert the bns commits like you suggested? Sorry if I created confusion on this but will let you handle in your AM.

**Comment by beguene** [comment]
Thanks @fbwoolf! It's not issue at all, we can indeed revert the bns commit. I will prepare a new PR for QA today.
Edit: I meant I will update this PR.

**Comment by beguene** [comment]
@Eshwari007 @timstackblock I updated this PR (and excluded the BNS fix), the PR description is also updated with the content of this release.
cc: @andresgalante

**Comment by Eshwari007** [comment]
The release is **blocked** due to https://github.com/hirosystems/stacks-wallet-web/issues/2235
cc @timstackblock @andresgalante

**Comment by timstackblock** [comment]
This issue is currently occurring on Silver Fox.
https://github.com/hirosystems/stacks-wallet-web/issues/2235

**Comment by andresgalante** [comment]
@Eshwari007 @timstackblock Please block the PR to prevent being merge when you are blocking a release

**Comment by beguene** [comment]
@timstackblock @Eshwari007 I just updated this release PR with the latest dev which includes the fix for #2235. I also updated the description.

**Review by andresgalante — CHANGES_REQUESTED** [review]
I am blocking this PR until Esh is done testing

**Comment by Eshwari007** [comment]
@markmhx Quick question more like enhancement:  

**Use case 1**
When a user logs into the wallet via secret key via the popup extension version, the accounts are populated accordingly but they are not on the full view window wallet
However if we sign in via the window wallet view ,the popup wallet version is synced up
Screen Shot 2022-02-16 at 2 41 43 PM



**Use Case 2**

Locking the account
Similarly when a user locks the account in the popup wallet version, both screens were locked but when the user unlocks the popup wallet version, the full window wallet is yet locked
Screen Shot 2022-02-17 at 9 56 19 AM

A few areas as above would be nice if they are synced up as the user needs to do duplicate work at times to view both screens concurrently. Thanks

**Comment by fbwoolf** [comment]
> 
> A few areas as above would be nice if they are synced up as the user needs to do duplicate work at times to view both screens concurrently. Thanks
>
This would definitely be an enhancement to routing using messaging. Currently, we only use messaging to keep them in sync on locking (not unlocking) and sign out.

**Comment by andresgalante** [comment]
I agree with Fara, I don't think these are blockers.

@Eshwari007 would you mind creating issue for your findings please?

**Comment by Eshwari007** [comment]
> I agree with Fara, I don't think these are blockers.
> 
> @Eshwari007 would you mind creating issue for your findings please?

Will do the needful and mark them as enhancements

**Review by Eshwari007 — APPROVED** [review]
Sanity test passed from QA's end . No major issues found. Enhancements and cosmetic bugs will be logged accordingly.

**Review by timstackblock — APPROVED** [review]
We can separate the issue in #2235 and go live with Silver Fox it looks like it needs a little more research and we shouldn't hold this PR indefinitely.

**Review by andresgalante — APPROVED** [review]
@Eshwari007 is ok to ship this release

**Comment by andresgalante** [comment]
@markmhx Can you please read #2235 and make sure that you are ok releasing `3.2` and introducing a minor bug. I don't think anyone will notice it.

**Comment by Eshwari007** [comment]
> @markmhx Can you please read #2235 and make sure that you are ok releasing `3.2` and introducing a minor bug. I don't think anyone will notice it.

Agree with @andresgalante on the above and we can ship the fix for the bug in the next release

**Comment by markmhendrickson** [comment]
Yep I'm okay with shipping Silver Fox without a fix for https://github.com/hirosystems/stacks-wallet-web/issues/2235 if we anticipate shipping that fix by the start of next week.
